packages/pwa/app/components/link/index.jsx

```diff
@@ -11,23 +11,43 @@ import {Link as SPALink, NavLink as NavSPALink} from 'react-router-dom'
 import {buildPathWithUrlConfig} from '../../utils/url'
 import useSite from '../../hooks/use-site'
 import useLocale from '../../hooks/use-locale'
+import {matchRoute} from '../../utils/routes-utils'
+import routes from '../../routes'
 
 const Link = React.forwardRef(({href, to, useNavLink = false, ...props}, ref) => {
     const _href = to || href
     const site = useSite()
     const locale = useLocale()
 
+    //TODO: Add configuration value to the default.js config defining the SFRA hostname
+    const urlHostname = 'https://sfra-site.com'
+
+    let _routes
+    if (typeof routes === 'function') {
+        _routes = routes()
+    }
+
+    // Remove route with * path
+    _routes = _routes.filter((_route) => _route.path !== '*')
```

Comment:
Is this expensive? Especially if we're doing it for many links on a page?

Reply:
The filtering is not necessary following your previous suggestion.

We'll still have one loop `routes.some()` which has an O(n) complexity and right now in the PWA n=493 routes including all the multi-site, multi-locale combinations.